hardware_interface/src/fake_components/generic_system.cpp

```diff
@@ -35,6 +36,38 @@ CallbackReturn GenericSystem::on_init(const hardware_interface::HardwareInfo & i
     return CallbackReturn::ERROR;
   }
 
+  auto set_nan_to_zero = [](auto param_1, auto param_2, auto & param_3) {
+    for (auto i = 0u; i < param_1.size(); i++)
+    {
+      for (auto j = 0u; j < param_2.size(); j++)
+      {
+        if (std::isnan(param_3[j][i]))
+        {
+          param_3[j][i] = 0.0;
+        }
+      }
+    }
+  };
+
+  auto populate_other_interfaces =
+    [](auto interface_list, auto standard_interfaces, auto & other_interfaces) {
+      for (const auto & interface : interface_list)
+      {
+        // add to list if non-standard interface
+        if (
+          std::find(standard_interfaces.begin(), standard_interfaces.end(), interface.name) ==
```

Comment:
this should also be populated using set

Reply:
I didn't understand what you tried to say here.

Reply:
you should use set here as also for the GPIOs